islandora_audio.module

```diff
@@ -210,6 +210,19 @@ function islandora_audio_islandora_sp_audiocmodel_islandora_ingest_steps() {
  * Implements hook_CMODEL_PID_islandora_derivative().
  */
 function islandora_audio_islandora_sp_audiocmodel_islandora_derivative() {
+  if (variable_get('islandora_audio_defer_derivatives_on_ingest', FALSE)) {
+    return array(
+      array(
+        'source_dsid' => NULL,
+        'destination_dsid' => 'TN',
+        'weight' => 1,
+        'function' => array(
+          'islandora_audio_create_thumbnail',
+        ),
+        'file' => drupal_get_path('module', 'islandora_audio') . '/includes/derivatives.inc',
+      ),
+    );
+  }
```

Comment:
I've made updates to the code to eliminate the duplicate TN array.  Please
let me know about other desired changes.
Thank you,
Robert

On Wed, Nov 4, 2015 at 10:55 AM, Jared Whiklo <notifications@github.com>
wrote:

> In islandora_audio.module
> <https://github.com/Islandora/islandora_solution_pack_audio/pull/123#discussion_r43897742>
> :
>
> > @@ -210,6 +210,19 @@ function islandora_audio_islandora_sp_audiocmodel_islandora_ingest_steps() {
> >   * Implements hook_CMODEL_PID_islandora_derivative().
> >   */
> >  function islandora_audio_islandora_sp_audiocmodel_islandora_derivative() {
> > +  if (variable_get('islandora_audio_defer_derivatives_on_ingest', FALSE)) {
> > +    return array(
> > +      array(
> > +        'source_dsid' => NULL,
> > +        'destination_dsid' => 'TN',
> > +        'weight' => 1,
> > +        'function' => array(
> > +          'islandora_audio_create_thumbnail',
> > +        ),
> > +        'file' => drupal_get_path('module', 'islandora_audio') . '/includes/derivatives.inc',
> > +      ),
> > +    );
> > +  }
>
> All and all this PR is fine with me, to avoid duplication of code. I would
> suggest instead of doing
>
> if (variable_get(...
> return array
> }
> return array
>
> You store the array and use your if (variable_get to either add or not
> add the PROXY_MP3 derivative to that array.
> If I am making any sense [image: :confused:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Islandora/islandora_solution_pack_audio/pull/123/files#r43897742>
> .
>



-- 
Robert Anderson
Software Engineer

6310 Hillside Court, Suite 101
Columbia, MD 21046 USA
Phone: 443.283.2137 <443.283.2120>
Fax: 443.283.2187 <443.283.2170>
Email: randerson@Quotient-Inc.com
Quotient-Inc.com